cli/command/container/opts.go

```diff
@@ -229,6 +230,7 @@ func addFlags(flags *pflag.FlagSet) *containerOptions {
 	flags.DurationVar(&copts.healthInterval, "health-interval", 0, "Time between running the check (ns|us|ms|s|m|h) (default 0s)")
 	flags.IntVar(&copts.healthRetries, "health-retries", 0, "Consecutive failures needed to report unhealthy")
 	flags.DurationVar(&copts.healthTimeout, "health-timeout", 0, "Maximum time to allow one check to run (ns|us|ms|s|m|h) (default 0s)")
+	flags.DurationVar(&copts.healthStartPeriod, "health-start-period", 0, "Start period for the container to initialize before starting health-retries countdown (ns|us|ms|s|m|h) (default 0s)")
```

Comment:
Can you add an annotation here so that it's hidden when talking to an older daemon (API version for [Docker 17.05 will be 1.29](https://github.com/docker/docker/blob/817106aaa1e8575d86e0b95a4219f6a7434572e3/api/common.go#L23-L24))?

```go
flags.SetAnnotation("health-start-period", "version", []string{"1.29"})
```